tuist/tuist
# (Prototype) Model Mappers

(Based on https://github.com/tuist/tuist/pull/1206)

### Short description 📝

Applying modifications to the `Project` and `Target` models within `GraphMappers` may not always be possible due to the current `Graph` structure. To create a mapper that can generate info plists for example, the both the `Target` and the hosting `Project` needs to be modified. 

### Solution 📦

Add "model mappers" that map `TuistCore` models ahead of converting them to a `Graph` representation.

### Implementation 👩‍💻👨‍💻

_Note: Those can be split into separate PRs to ease the review process but grouped here to demo the concept_

- [x] Add model mappers
- [x] Modify loading process to accommodate model mappers
- [x] Add InfoPlist mapper
- [ ] Model the current [`enriched(model: TuistCore.Project, with config: TuistCore.Config)`](https://github.com/tuist/tuist/blob/master/Sources/TuistLoader/Loaders/GeneratorModelLoader.swift#L52) as model mapper

### Test Plan 🛠

- Run `tuist generate` within `fixtures/ios_app_with_watchapp2`
- Verify the Info plist files are generated in the `Derived` directory 
- Open `App.xcworkspace`
- Verify the info plists are associated/assigned for the watch app and watch extension target
